src/python/pants/backend/python/typecheck/mypy/rules.py

```diff
@@ -92,13 +101,9 @@ def generate_args(mypy: MyPy, *, file_list_path: str) -> Tuple[str, ...]:
         import site
         import sys
 
-        PYTHONPATH = frozenset(
-            os.path.realpath(p)
-            for p in os.environ.get('PYTHONPATH', '').split(os.pathsep)
-        )
```

Comment:
I removed this part because I don't think it's relevant in a v2 world with hermetic environments. Users can't set `PYTHONPATH` manually anymore. I think this was leftover from the original v1 script.

Lmk though if this needs to be added back.